server/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -405,9 +402,8 @@ protected void doStop() {
             });
         }
         try {
-            if (latch.await(shardsClosedTimeout.seconds(), TimeUnit.SECONDS) == false) {
-                logger.warn("Not all shards are closed yet, waited {}sec - stopping service", shardsClosedTimeout.seconds());
-            }
+            // infinite wait
```

Comment:
nit: no need for this comment here IMO, it's obviously a wait without a timeout now :)